File: frontend/src/service/ocp-express/InfoController.ts
/*
 * Copyright (c) 2023 OceanBase
 * OCP Express is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *          http://license.coscl.org.cn/MulanPSL2
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 */

/* eslint-disable */
// 该文件由 OneAPI 自动生成，请勿手动修改！
import request from '@/util/request';

/** 此处后端没有提供注释 GET /api/v1/druid/stat */
export async function druidStat(options?: { [key: string]: any }) {
  return request<API.OneApiResult_object_>('/api/v1/druid/stat', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/git-info */
export async function gitInfo(options?: { [key: string]: any }) {
  return request<API.Map_String_Object_>('/api/v1/git-info', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/ */
export async function home(options?: { [key: string]: any }) {
  return request<API.OneApiResult_string_>('/api/v1/', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/info */
export async function info(options?: { [key: string]: any }) {
  return request<API.Map_String_Object_>('/api/v1/info', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/status */
export async function status(options?: { [key: string]: any }) {
  return request<API.Map_String_Object_>('/api/v1/status', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /api/v1/time */
export async function time(options?: { [key: string]: any }) {
  return request<API.OneApiResult_offsetdatetime_>('/api/v1/time', {
    method: 'GET',
    ...(options || {}),
  });
}

File: server/src/main/java/com/oceanbase/ocp/controller/InfoController.java
/*
 * Copyright (c) 2023 OceanBase
 * OCP Express is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *          http://license.coscl.org.cn/MulanPSL2
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 */
package com.oceanbase.ocp.controller;

import com.oceanbase.ocp.common.util.HostUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.info.BuildProperties;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.OffsetDateTime;
import java.time.ZoneId;
import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/v1")
public class InfoController {

    @Autowired
    private BuildProperties buildProperties;

    @GetMapping("/")
    public String home() {
        return "index.html";
    }

    @GetMapping("/status")
    public Map<String, Object> status() {
        Map<String, Object> clientInfo = new HashMap<>(8);
        clientInfo.put("status", "ok");
        clientInfo.put("version", buildProperties.getVersion());
        clientInfo.put("server", HostUtils.getLocalIp());
        return clientInfo;
    }

    @GetMapping("/time")
    public OffsetDateTime time() {
        return OffsetDateTime.now();
    }

    @GetMapping("/info")
    public Map<String, Object> info() {
        Map<String, Object> infoMap = new HashMap<>();
        infoMap.put("name", buildProperties.getName());
        infoMap.put("group", buildProperties.getGroup());
        infoMap.put("artifact", buildProperties.getArtifact());
        infoMap.put("buildTime", OffsetDateTime.ofInstant(buildProperties.getTime(), ZoneId.systemDefault()));
        infoMap.put("buildVersion", buildProperties.getVersion());
        infoMap.put("buildJavaVersion", buildProperties.get("java.version"));
        infoMap.put("springBootVersion", buildProperties.get("spring-boot.version"));
        infoMap.put("server", HostUtils.getLocalIp());
        return infoMap;
    }

}
